Minio: Unable to locate credentials

<!-- Thanks for submitting an issue! Please describe the issue. -->

# Expectation / Proposal

Hi, when I use S3 Bucket (Minio) on prefect I got an error like:

```
botocore.exceptions.NoCredentialsError: Unable to locate credentials
```

After some investigation, I find that this line:

https://github.com/PrefectHQ/prefect-aws/blob/7a17997de8a61ad977549ed77e0ca6f669492d31/prefect_aws/s3.py#L305

If I change this line into:

```python
credentials: Optional[MinIOCredentials] = Field(
```

Things will be Ok.

Blocks:

![image](https://user-images.githubusercontent.com/45716019/218293969-94758a08-7853-43c7-a5aa-0bba67c142f7.png)

My python version: Python 3.10.10
prefect version: ==2.8.0
prefect-aws version: commit 7a17997de8a61ad977549ed77e0ca6f669492d31

# Traceback / Example

```python
from prefect import flow, task
from prefect_aws.s3 import S3Bucket
from prefect_aws import MinIOCredentials

@flow
def my_favorite_function():
    minio_credentials_block = MinIOCredentials.load("testblock")
    s3_bucket_block = S3Bucket(
        bucket_name="testblock",
        minio_credentials=minio_credentials_block
    )
    print(s3_bucket_block.list_objects(""))
    return "test"

print(my_favorite_function())
```

- [ ] I would like to [help contribute](https://PrefectHQ.github.io/prefect-aws/#contributing) a pull request to resolve this!

Error in prefect-aws==0.3.2:
Register blocks:
```python
@sync_compatible
async def save_conf_blocks():
  endpoint_url = ENDPOINT_URL
  minio = MinIOCredentials.parse_obj(
    dict(name='epool-tk-upd-minio', minio_root_user=environ['MINIO_LOGIN'],
      minio_root_password=environ['MINIO_PASSW'],
      aws_client_parameters=dict(endpoint_url=endpoint_url), ))
  await minio.save('epool-tk-upd-minio', overwrite=True)

  s3_bucket_block = S3Bucket(
    name='epool-tk-upd', bucket_name='tk-upd-cache',
    credentials=minio, # minio_credentials=minio,
    endpoint_url=endpoint_url, )
  await s3_bucket_block.save('epool-tk-upd', overwrite=True)
```

Flow code:
```python
@task(cache_result_in_memory=False, persist_result=False, log_prints=True)
async def save_result(cache: WritableFileSystem, path, data):
  assert (
    isinstance(cache, LocalFileSystem) or
    isinstance(cache.credentials, MinIOCredentials)
  ), f'{cache}: {cache.credentials}'
  await cache.write_path(path, out_data)

@sync_compatible
async def s3_bucket_block() -> S3Bucket:
  _s3_bucket_block = await S3Bucket.load('epool-tk-upd')
  return _s3_bucket_block

@flow
def tk_upd_cache():
  cache=s3_bucket_block()
  save_result.submit(cache, 'tk_name', b'opa')
```

Error out:
```
Encountered exception during execution:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/prefect/engine.py", line 1623, in orchestrate_task_run
    result = await call.aresult()
             ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/prefect/_internal/concurrency/calls.py", line 181, in aresult
    return await asyncio.wrap_future(self.future)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/prefect/_internal/concurrency/calls.py", line 218, in _run_async
    result = await coro
             ^^^^^^^^^^
  File "/prj/flows/tools.py", line 179, in save_result
    assert (
AssertionError: S3Bucket(bucket_name='tk-upd-cache', credentials=AwsCredentials(aws_access_key_id=None, aws_secret_access_key=None, aws_session_token=None, profile_name=None, region_name='ru-central1', aws_client_parameters=AwsClientParameters(api_version=None, use_ssl=True, verify=True, verify_cert_path=None, endpoint_url='https://s3.epool.ru', config=None)), bucket_folder=''): AwsCredentials(aws_access_key_id=None, aws_secret_access_key=None, aws_session_token=None, profile_name=None, region_name='ru-central1', aws_client_parameters=AwsClientParameters(api_version=None, use_ssl=True, verify=True, verify_cert_path=None, endpoint_url='https://s3.epool.ru', config=None))
01:43:44 PM
save_result-0
prefect.task_runs
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Bug where `S3Bucket.load()` constructed `AwsCredentials` instead of `MinIOCredentials` - [#359](https://github.com/PrefectHQ/prefect-aws/pull/359)
+
 ### Deprecated
 
 ### Removed

--- a/prefect_aws/s3.py
+++ b/prefect_aws/s3.py
@@ -412,7 +412,7 @@ class S3Bucket(WritableFileSystem, WritableDeploymentStorage, ObjectStorageBlock
 
     bucket_name: str = Field(default=..., description="Name of your bucket.")
 
-    credentials: Union[AwsCredentials, MinIOCredentials] = Field(
+    credentials: Union[MinIOCredentials, AwsCredentials] = Field(
         default_factory=AwsCredentials,
         description="A block containing your credentials to AWS or MinIO.",
     )
@@ -424,9 +424,6 @@ class S3Bucket(WritableFileSystem, WritableDeploymentStorage, ObjectStorageBlock
             "for reading and writing objects."
         ),
     )
-
-    class Config:
-        smart_union = True
 
     # Property to maintain compatibility with storage block based deployments
     @property

--- a/tests/test_s3.py
+++ b/tests/test_s3.py
@@ -822,7 +822,10 @@ class TestS3Bucket:
 
     def test_credentials_are_correct_type(self, credentials):
         s3_bucket = S3Bucket(bucket_name="bucket", credentials=credentials)
+        s3_bucket_parsed = S3Bucket.parse_obj({"bucket_name": "bucket", "credentials": dict(credentials)})
         assert isinstance(s3_bucket.credentials, type(credentials))
+        assert isinstance(s3_bucket_parsed.credentials, type(credentials))
+
 
     @pytest.mark.parametrize("client_parameters", aws_clients[-1:], indirect=True)
     def test_list_objects_empty(self, s3_bucket_empty, client_parameters):

--- a/tests/test_s3.py
+++ b/tests/test_s3.py
@@ -822,10 +822,11 @@ class TestS3Bucket:
 
     def test_credentials_are_correct_type(self, credentials):
         s3_bucket = S3Bucket(bucket_name="bucket", credentials=credentials)
-        s3_bucket_parsed = S3Bucket.parse_obj({"bucket_name": "bucket", "credentials": dict(credentials)})
+        s3_bucket_parsed = S3Bucket.parse_obj(
+            {"bucket_name": "bucket", "credentials": dict(credentials)}
+        )
         assert isinstance(s3_bucket.credentials, type(credentials))
         assert isinstance(s3_bucket_parsed.credentials, type(credentials))
-
 
     @pytest.mark.parametrize("client_parameters", aws_clients[-1:], indirect=True)
     def test_list_objects_empty(self, s3_bucket_empty, client_parameters):

